lib/pool.dart

```diff
@@ -121,6 +121,44 @@ class Pool {
     });
   }
 
+  /// Requests a resource for the combined duration of the [callback] and the
+  /// processing of the Stream it returns.
+  /// (E.g. a database connection is hold until the the results get processed.)
+  Stream/*<T>*/ streamWithResource/*<T>*/(Stream/*<T>*/ callback()) {
+    StreamController/*<T>*/ streamController;
+    StreamSubscription/*<T>*/ streamSubscription;
+    Completer resourceCompleter = new Completer();
+    final complete = () {
+      if (streamSubscription != null) {
+        streamSubscription.cancel();
+        streamSubscription = null;
+      }
+      if (!resourceCompleter.isCompleted) {
+        resourceCompleter.complete();
+        streamController.close();
+      }
+    };
+    streamController = new StreamController/*<T>*/(onCancel: complete);
+    withResource(() {
```

Comment:
I think we need similar error handling as we have around the `withResource()`, and using `request()` will just duplicate it (adding a few more lines of code). E.g. the `resource.release()` should be placed in `complete` to get it consistent with the stream handler -> separate field definition, value allocation etc.
If you think it is worth it, I would do it, but I like it this way better.